pkgs/by-name/re/renovate/package.nix

```diff
@@ -0,0 +1,93 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  nodePackages,
+  nodejs,
+  pnpm_9,
+  python3,
+  renovate,
+  testers,
+  xcbuild,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "renovate";
+  version = "37.393.0";
+
+  src = fetchFromGitHub {
+    owner = "renovatebot";
+    repo = "renovate";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-YgxcGNMgmwrausdR7kvG1NiyQPn0FcCq/isf9qUDCFY=";
+  };
+
+  postPatch = ''
+    substituteInPlace package.json \
+      --replace-fail "0.0.0-semantic-release" "${finalAttrs.version}"
+  '';
+
+  nativeBuildInputs = [
+    makeWrapper
+    nodePackages.node-gyp
+    nodejs
+    pnpm_9.configHook
+    python3
+  ] ++ lib.optional stdenv.hostPlatform.isDarwin xcbuild;
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-Zbe561q6xDKDIN+E/2eyQMz2GtpPvJEv2pAauMa+8pE=";
+  };
+
+  env.COREPACK_ENABLE_STRICT = 0;
+
+  buildPhase = ''
+    runHook preBuild
+
+    pnpm build
+
+    rm -r node_modules
+    pnpm install --prod --offline --ignore-scripts
+
+    pushd node_modules/.pnpm/re2*/node_modules/re2
+
+    mkdir -p $HOME/.node-gyp/${nodejs.version}
+    echo 9 > $HOME/.node-gyp/${nodejs.version}/installVersion
+    ln -sfv ${nodejs}/include $HOME/.node-gyp/${nodejs.version}
+    export npm_config_nodedir=${nodejs}
+    npm run rebuild
```

Comment:
could you add a comment explaining why this is required?

Reply:
updated https://github.com/NixOS/nixpkgs/compare/0ccf4380c5bd466993ae3853d8a05c757a5ecd17..f324b0fecd339ef697c8c9a05fdfbeb272f57568